cdap-system-app-api/src/main/java/io/cdap/cdap/api/service/http/SystemHttpServiceContext.java

```diff
@@ -61,4 +61,15 @@
    */
   Map<String, String> evaluateMacros(String namespace, Map<String, String> properties,
                                      MacroEvaluator evaluator, MacroParserOptions options) throws InvalidMacroException;
+
+  /**
+   * Get preferences for the given entity
+   *
+   * @param entityId the id of the entity to fetch preferences for
+   * @param resolved true if resolved properties are desired.
+   * @return Map containing Preferences keys and values.
+   */
+  default Map<String, String> getPreferences(EntityId entityId, boolean resolved) {
```

Comment:
Better not to add cdap-proto as a dependency, we try to keep the api modules free of any dependencies except other api modules.

Here I think it's better to just take the namespace as a String argument, as it makes it more similar to the existing APIs, and we don't currently have a use case for not resolving and for getting other entities. I think it also would be better to declare that it throws IOException so that callers are forced to deal with the fact that this is a network call that may fail.

Reply:
I'll remove the dependency to Cdap proto and rename the method to make it obvious that this is a call to fetch preferences for a given namespace.